# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1389118693 18000
#      Tue Jan 07 13:18:13 2014 -0500
# Branch gui-release
# Node ID 2eb26867bf66f979c55f9dbc1f3dcbbb966961e8
# Parent  9a0f996a358ad53ca95818ac69f76170c3f76a97
update COMMAND_START pattern in lexer

* lex.ll (<COMMAND_START>(\.|[^#% \t\r\n\.\,\;\"\'\(\[\{\}\]\)]+)):
Fix pattern to not allow '.' to be followed by another '.'.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -417,18 +417,18 @@ ANY_INCLUDING_NL (.|{NL})
 
     curr_lexer->current_input_column += yyleng;
   }
 
 %{
 // Everything else is slurped into the command arguments.
 %}
 
-<COMMAND_START>([\.]|[^#% \t\r\n\,\;\"\'\(\[\{\}\]\)]+) {
-    curr_lexer->lexer_debug ("<COMMAND_START>[^#% \\t\\r\\n\\.\\,\\;\\\"\\'\\(\\[\\{\\}\\]\\)]+");
+<COMMAND_START>(\.|[^#% \t\r\n\.\,\;\"\'\(\[\{\}\]\)]+) {
+    curr_lexer->lexer_debug ("<COMMAND_START>(\\.|[^#% \\t\\r\\n\\.\\,\\;\\\"\\'\\(\\[\\{\\}\\]\\)]+)");
 
     curr_lexer->string_text += yytext;
     curr_lexer->current_input_column += yyleng;
   }
 
 <MATRIX_START>{S}* {
     curr_lexer->lexer_debug ("<MATRIX_START>{S}*");
 
